server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachmentTest.java

```diff
@@ -0,0 +1,243 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mailet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.apache.mailet.base.test.MimeMessageBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TextCalendarBodyToAttachmentTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private TextCalendarBodyToAttachment mailet = new TextCalendarBodyToAttachment();
+
+    private MimeMessage calendarMessage;
+
+    @Before
+    public void setUp() throws Exception {
+        mailet.init(FakeMailetConfig.builder()
+            .mailetName("TextCalendarBodyToAttachment")
+            .build());
+
+        calendarMessage = MimeMessageBuilder.mimeMessageFromStream(ClassLoader.getSystemResourceAsStream("calendar.eml"));
+    }
+
+    @Test
+    public void getMailetInformationShouldReturnInformation() throws Exception {
+        assertThat(mailet.getMailetInfo()).isEqualTo("Moves body part of content type text/calendar to attachment");
+    }
+
+    @Test
+    public void serviceShouldThrowWhenCanNotGetMessageFromMail() throws Exception {
+        expectedException.expect(MessagingException.class);
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenThrow(MessagingException.class);
+
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenMessageCanNotGetContentType() throws Exception {
+        expectedException.expect(MessagingException.class);
+
+        MimeMessage message = mock(MimeMessage.class);
+        Mail mail = FakeMail.from(message);
+
+        when(message.isMimeType(anyString())).thenThrow(MessagingException.class);
+
+        mailet.service(mail);
```

Comment:
I'm not sure we should throw on such malformed mails

Reply:
We did not validate Mail before process on any mailet as I saw. Which kind of validations that we expect? Mail must have from, to ...

Reply:
no I don't expect to add validation, just that if the mailet reads such an email, it ignores it instead of throwing

Reply:
Yes, it's ignore